src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/directionsquestionnaire/HearingContentProvider.java

```diff
@@ -0,0 +1,88 @@
+package uk.gov.hmcts.cmc.claimstore.documents.content.directionsquestionnaire;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.claimstore.services.staff.models.ExpertReportContent;
+import uk.gov.hmcts.cmc.claimstore.services.staff.models.HearingContent;
+import uk.gov.hmcts.cmc.claimstore.utils.DateUtils;
+import uk.gov.hmcts.cmc.claimstore.utils.Formatting;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.DirectionsQuestionnaire;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.ExpertReport;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.ExpertRequest;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.RequireSupport;
+import uk.gov.hmcts.cmc.domain.models.directionsquestionnaire.UnavailableDate;
+import uk.gov.hmcts.cmc.domain.models.response.YesNoOption;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.cmc.ccd.util.StreamUtil.asStream;
+
+@Component
+public class HearingContentProvider {
+
+    private static final String DISABLED_ACCESS = "Disabled Access";
+    private static final String YES = "YES";
+    private static final String NO = "NO";
+
+    private Function<ExpertReport, ExpertReportContent> mapExpertReport = report ->
+        ExpertReportContent.builder().expertName(report.getExpertName())
+            .expertReportDate(Formatting.formatDate(report.getExpertReportDate()))
+            .build();
+    private Function<UnavailableDate, String> mapToISOFullStyle = unavailableDate ->
```

Comment:
maybe I am missing something, but why can't these functions be simple methods? ie. `String mapToISOFullStyle(UnavailableDate unavailableDate)`? 😳 (and similarly for mapExpertReport)

Reply:
Function, Supplier, Consumer are predefined interface. Its strict definition enforces input and return types. So its better to use predefined FI. Everything can be done thru normal method. But having method adhering to a interface (Definition) is much more cleaner IMHO.

Reply:
I hear you, but I'd say a method definition is equivalent to a `Function`, as in it still enforces a return type and the parameters types. I'd use a Function if I needed to pass that as a parameter to another function, or store it in a map, or compose it / do higher level operations on it, but if all you're doing is calling it, I feel it's just less readable.